From 2bb665209167463d83dd9b7ef18ad3791562bdd9 Mon Sep 17 00:00:00 2001
From: Mythri P K <mythri.p.k@intel.com>
Date: Fri, 18 Jul 2014 19:23:35 +0530
Subject: [PATCH 20/39] ASoC: soc-core: In volsw_info set min value to
 mixer_control struct min

Issue: GMINL-3679

In case of controls like  gain the min value can be negative, they
still use the common volsw info, so set min value using mc->min
than by setting to 0 by default.

Change-Id: I5b5c77106d5690f330579cf8eddb83adb3d79cd9
Signed-off-by: Mythri P K <mythri.p.k@intel.com>
---
 sound/soc/soc-core.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/sound/soc/soc-core.c b/sound/soc/soc-core.c
index d54b0fa..f20cc5a 100644
--- a/sound/soc/soc-core.c
+++ b/sound/soc/soc-core.c
@@ -2807,7 +2807,7 @@ int snd_soc_info_volsw(struct snd_kcontrol *kcontrol,
 		uinfo->type = SNDRV_CTL_ELEM_TYPE_INTEGER;
 
 	uinfo->count = snd_soc_volsw_is_stereo(mc) ? 2 : 1;
-	uinfo->value.integer.min = 0;
+	uinfo->value.integer.min = mc->min;
 	uinfo->value.integer.max = platform_max - mc->min;
 	return 0;
 }
-- 
1.9.1

